sdk/src/lib.rs

```diff
@@ -68,7 +68,8 @@ pub mod example_mocks;
 pub mod exit;
 pub mod feature;
 pub mod genesis_config;
-pub mod hard_forks;
+#[deprecated(since = "2.2.0", note = "Use `solana-hard-forks` crate instead")]
```

Comment:
I intentionally took this out. Maybe I'm missing something on what `full` is for?
https://github.com/anza-xyz/agave/pull/3855#pullrequestreview-2483235148

Reply:
The "full" feature is a historical artifact from when programs used `solana-sdk`. The concept is that on-chain programs used `solana-sdk` without the "full" feature enabled, and off-chain packages would use `solana-sdk` with the "full" feature.

So by not gating on the "full" feature, this PR adds `HardForks` into non-full builds. It's not a breaking change since it looks like an addition, but it is a change from the current crate.

More info at https://github.com/solana-labs/solana/pull/12989

Reply:
Ok cool thanks, I originally had this under `full` and was following a similar understanding based on the goals of the SDK vs. program SDK extraction plan. Added back.